nupic/algorithms/anomaly.py

```diff
@@ -96,10 +96,14 @@ def __init__(self, slidingWindowSize=None, mode=MODE_PURE, binaryAnomalyThreshol
 
     if self._mode == Anomaly.MODE_LIKELIHOOD or self._mode == Anomaly.MODE_WEIGHTED:
       self._likelihood = AnomalyLikelihood() # probabilistic anomaly
+    else:
+      self._likelihood = None
+
     if not self._mode in Anomaly._supportedModes:
```

Comment:
I do not expect the modes[] to change, but you're probably right, it's not marked as static..